src/main/java/com/ita/if103java/ims/dao/impl/DashboardDaoImpl.java

```diff
@@ -102,50 +102,50 @@ public WarehousePremiumStructDto getPreLoadByAccounId(Long id, Long accountId) {
-                wpld.getId(), accountId, accountId, accountId, accountId);
-            wpld.setCapacity(cc.getCapacity());
-            wpld.setCharge(cc.getCharge());
-            wpld.setChilds(findSub(id,wpld.getLevel(), accountId));
+
+            wpld.setChilds(findSub(id, id, wpld.getLevel(), accountId));
+
+            recFill(wpld);
+
             return wpld;
         } catch (DataAccessException e) {
-            throw crudException(e.toString(),"getPreLoad","accountId=" + accountId);
+            throw crudException(e,"getPreLoad","accountId=" + accountId);
         }
     }
 
-    private List<WarehousePremiumStructDto> findSub(Long id, int level, Long accountId){
+    private List<WarehousePremiumStructDto> findSub(Long id, Long top_warehouse_id, int level, Long accountId){
         try{
             List<WarehousePremiumStructDto> wpld = jdbcTemplate.query(Queries.SQL_WAREHOUSE_STRUCTURE_SUB,
                 warehousePremiumStructRowMapper, id, accountId);
-                for (int i = 0; i < wpld.size(); i++) {
-                    wpld.get(i).setLevel(level + 1);
-
-                    ChargeCapacity cc1 = jdbcTemplate.queryForObject(Queries.SQL_FIND_SUBS_LOAD,
-                        chargeCapacityRowMapper,
-                        wpld.get(i).getId(), accountId, accountId, accountId, accountId);
-
-                    wpld.get(i).setCharge(cc1.getCharge());
-                    wpld.get(i).setCapacity(cc1.getCapacity());
-
-                    wpld.get(i).setChilds(findSub(wpld.get(i).getId(), wpld.get(i).getLevel(), accountId));
-                    if (wpld.get(i).getChilds().size() == 0){
-                        ChargeCapacity cc2 = jdbcTemplate.queryForObject(Queries.SQL_FIND_BOT_LOAD,
-                            chargeCapacityRowMapper, wpld.get(i).getId(), accountId);
-
-                        wpld.get(i).setCharge(cc2.getCharge());
-                        wpld.get(i).setCapacity(cc2.getCapacity());
-                    }
+            for (int i = 0; i < wpld.size(); i++) {
+                wpld.get(i).setLevel(level + 1);
+                wpld.get(i).setChilds(findSub(wpld.get(i).getId(), top_warehouse_id, wpld.get(i).getLevel(), accountId));
+                if (wpld.get(i).getChilds().size()==0){
+                    ChargeCapacity chargeCapacity = jdbcTemplate.queryForObject(Queries.SQL_FIND_BOT_CAPACITY_CHARGE,
+                        chargeCapacityRowMapper, top_warehouse_id, wpld.get(i).getId(), accountId);
+                    wpld.get(i).setCharge(chargeCapacity.getCharge());
+                    wpld.get(i).setCapacity(chargeCapacity.getCapacity());
                 }
-                return wpld;
+            }
+            return wpld;
         } catch (DataAccessException e) {
-            throw crudException(e.toString(),"findSub","*");
+            throw crudException(e,"findSub","*");
+        }
+    }
+
+    private void recFill(WarehousePremiumStructDto wp){
```

Comment:
recFill - please do not shorten methods and variables, because your code is hard to read and understand